File: portal/portal/App_Code/DAO/MenuItemDAO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MySql.Data.MySqlClient;
using System.IO;

namespace portal.App_Code.DAO
{
    public class MenuItemDAO
    {
        //lendo string de conexão     
        String sc = Properties.Settings.Default.CN;
        String logPath = Properties.Settings.Default.PathErrorLog;

        public MenuItem carregar(long pCodigo)
        {
            MenuItem obj = new MenuItem();
            try
            {
                //cria a conexao com o bd
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Menu_Item where codigo=" + pCodigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                if (registros.Read())
                {
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.CodigoMenu = Convert.ToInt64(registros["codigo_Menu"]);
                    obj.CodigoItemPai = Convert.ToInt64(registros["codigo_item_pai"]);
                    obj.Nome = registros["senha"].ToString();
                    obj.CodigoConteudo = Convert.ToInt64(registros["codigo_conteudo"]);
                    obj.Url = registros["url"].ToString();
                }
                conexao.Close();
            }
            catch (Exception err)
            {
                String log = "Erro=>" + DateTime.Now + err.Message + Environment.NewLine;
                File.AppendAllText(logPath, log);
            }
            return obj;
        }

        public List<MenuItem> carregarLista(String[] pFiltros, String pOrdena)
        {
            List<MenuItem> lista = new List<MenuItem>();
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Menu_Item where ";
                for (int i = 0; i < pFiltros.Length; i++)
                {
                    sql = sql + pFiltros[i];
                }
                sql = sql + " order by " + pOrdena;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    MenuItem obj = new MenuItem();
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.CodigoMenu = Convert.ToInt64(registros["codigo_Menu"]);
                    obj.CodigoItemPai = Convert.ToInt64(registros["codigo_item_pai"]);
                    obj.Nome = registros["senha"].ToString();
                    obj.CodigoConteudo = Convert.ToInt64(registros["codigo_conteudo"]);
                    obj.Url = registros["url"].ToString();
                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }

        public List<MenuItem> carregarLista(int codigo)
        {
            List<MenuItem> lista = new List<MenuItem>();
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Menu_Item where codigo = " + codigo;
                
                sql = sql + " order by " + codigo ;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    MenuItem obj = new MenuItem();
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.CodigoMenu = Convert.ToInt64(registros["codigo_Menu"]);
                    obj.CodigoItemPai = Convert.ToInt64(registros["codigo_item_pai"]);
                    obj.Nome = registros["senha"].ToString();
                    obj.CodigoConteudo = Convert.ToInt64(registros["codigo_conteudo"]);
                    obj.Url = registros["url"].ToString();
                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }

        public static List<MenuItem> carregarLista(double pCodigoMenu)
        {
            String sc = Properties.Settings.Default.CN;
            String logPath = Properties.Settings.Default.PathErrorLog;
            List<MenuItem> lista = new List<MenuItem>();
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Menu_Item where codigo_menu=" + pCodigoMenu.ToString();
                
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    MenuItem obj = new MenuItem();
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.CodigoMenu = Convert.ToInt64(registros["codigo_Menu"]);
                    obj.CodigoItemPai = Convert.ToInt64(registros["codigo_item_pai"]);
                    obj.Nome = registros["senha"].ToString();
                    obj.CodigoConteudo = Convert.ToInt64(registros["codigo_conteudo"]);
                    obj.Url = registros["url"].ToString();
                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }

        public void persistir(MenuItem obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "";
                if (obj.Codigo == 0)
                {
                    sql = "insert into Menu_Item(nome,codigo_menu, codigo_item_pai, codigo_conteudo, url) value('{0}',{1},{2},{3},'{4}')";
                    sql = String.Format(sql, obj.Nome, obj.CodigoMenu, obj.CodigoItemPai, obj.CodigoConteudo, obj.Url);
                }
                else
                {
                    sql = "update Menu_Item set nome='{0}', codigo_menu={1}, codigo_item_pai={2}, codigo_conteudo={3},url='{4}'  where codigo={5}";
                    sql = String.Format(sql, obj.Nome, obj.CodigoMenu, obj.CodigoItemPai, obj.CodigoConteudo, obj.Url, obj.Codigo);   
                }

                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }


        public void remover(MenuItem obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "delete from Menu_Item where codigo=" + obj.Codigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }

    }
}

File: portal/portal/App_Code/DAO/MenuItemDAO.php
<?php
namespace portal\App_Code\DAO {
//CHAMADAS EXTERNAS------------------------------------------------------------------------------------------------------------------------------
use portal\App_Code\MenuItem;
mysqli_report(MYSQLI_REPORT_STRICT|MYSQLI_REPORT_ERROR);     

    try{
        require_once("/../MenuItem.php");        
        require_once("/../../connection.php");        
    }catch(Exception $e){
		require_once (dirname(__file__, 2) . "\MenuItem.php");
		require_once (dirname(__file__, 3) . "\connection.php");
    }
	
class MenuItemDAO{
//CONSTRUTOR----------------------------------------------------------------------------------------------------------
	public function __construct (){}
//INSERIR---------------------------------------------------------------------------------------------------------------------------------------- 
    function inserirMenuItem($MenuItem){
        try{
            $con = BD_AbrirConexao();
            $sql = "insert into menu_item (codigo_menu, codigo_item_pai,nome,codigo_conteudo,url) values (' " .
                 $MenuItem->getCodigoMenu() . " ',' " .
                 $MenuItem->getCodigoItemPai() . " ',' " .
				 $MenuItem->getNome() . " ',' " .
                 $MenuItem->getCodigoConteudo() . " ',' " .
                 $MenuItem->getURL() . " ')";
                 mysqli_query($con,$sql);
                 echo "Inserção com sucesso!";
        }catch (Exception $e){
                echo "Erro de inserção: " . $e;
        }
    mysqli_close($con);	
    }
//ALTERAR--------------------------------------------------------------------------------------------------------------------------------------   
    function alterarMenuItem($MenuItem){  
        try{
            $con = BD_AbrirConexao();
            $sql = "update menu_item set " . 
                "codigo_menu = ' " . $MenuItem->getCodigoMenu() . "', " .
                "codigo_item_pai = ' " . $MenuItem->getCodigoItemPai() . "', " .
                "nome = ' " . $MenuItem->getNome() . "', " .
                "codigo_conteudo = ' " . $MenuItem->getCodigoConteudo() . "', " .
                "url = ' " . $MenuItem->getURL() . "' " .
                "where codigo = " . $MenuItem->getCodigo();
                mysqli_query($con,$sql);
                echo "Dados alterados com sucesso";
        }catch (Exception $e){
                echo "Erro ao alterar dados: " . $e;
                }
    mysqli_close($con);	
    }
//CARREGAR--------------------------------------------------------------------------------------------------------------------------------------
    function carregarMenuItem ($codigo){
            try {
				$sql = "select * from menu_item where codigo = " . $codigo;
                $con = BD_AbrirConexao();
                $gargantula = mysqli_query($con,$sql);
                    if($carga = mysqli_fetch_array($gargantula)){
                        $MenuItem = new MenuItem();
                        $MenuItem->setMenuItem(($carga["codigo"]),
                                               ($carga["codigo_menu"]),
                                               ($carga["codigo_item_pai"]),
                                               ($carga["nome"]),
                                               ($carga["codigo_conteudo"]),
                                               ($carga["url"]));
                        mysqli_close($con);
                        return $MenuItem;
                    }else{
                        echo "Registros não encontrados!";
                    }
            }catch (Exception $e){
                echo "Erro de busca: " . $e;
            }
        mysqli_close($con);
    }
        
//CARREGA LISTA DE MENU ITEM MODO 1 -----------------------------------------------------------------------------------------------------
    function carregarListaMenuItem($qtdOpcoes, $ordem){
        $sql = "select * from menu_item where ";
        
            for ($i = 0 ; $i <= strlen((array)$qtdOpcoes) ; $i++){
                $sql = $sql . $qtdOpcoes[$i];
                    if($i != $qtdOpcoes[$i] - 1){
                        $sql = $sql . "and ";  
                    }
                $sql = $sql . " order by " . $ordem;
                
                try{
                    $con = BD_AbrirConexao();
                    $gargantula = mysqli_query($con,$sql);                
                    if($carga = mysqli_fetch_array($gargantula)){
                       $listaMenuItem = [];
                        while($carga = mysqli_fetch_array($gargantula)){
                            $menu_item = new portal\App_Code\MenuItem();
                            $menu_item->setMenuItem(($carga["codigo"]),
                                               ($carga["codigo_menu"]),
                                               ($carga["codigo_item_pai"]),
                                               ($carga["nome"]),
                                               ($carga["codigo_conteudo"]),
                                               ($carga["url"]));
                            $listaMenuItem[] = $menu_item;
                        }
                        mysqli_close($con);
                        return $listaMenuItem;
                    }else{
                        echo "Registros não encontrados!";
                    }
                }catch (Exception $e){
                echo "Erro de busca: " . $e;           
            }
            mysqli_close($con);
            }
    }
//CARREGA LISTA DE MENU ITEM MODO 2 -----------------------------------------------------------------------------------------------------
    function carregarListaMenuItemCod($dado){
                try{
					$sql = $sql = "select * from menu_item where codigo_menu = " . $dado;
                    $con = BD_AbrirConexao();
                    $gargantula = mysqli_query($con,$sql);                
                    if($carga = mysqli_fetch_array($gargantula)){
                       $listaMenuItem = [];
                        while($carga = mysqli_fetch_array($gargantula)){
                            $menu_item = new portal\App_Code\MenuItem();
                            $menu_item->setMenuItem(($carga["codigo"]),
                                               ($carga["codigo_menu"]),
                                               ($carga["codigo_item_pai"]),
                                               ($carga["nome"]),
                                               ($carga["codigo_conteudo"]),
                                               ($carga["url"]));
                            $listaMenuItem[] = $menu_item;
                        }
                        mysqli_close($con);
                        return $listaMenuItem;
                    }else{
                        echo "Registros não encontrados!";
                    }
                }catch (Exception $e){
                echo "Erro de busca: " . $e;           
            }
            mysqli_close($con);
            }
    

//DELETAR----------------------------------------------------------------------------------------------------------------------------------------
    function deletarMenuItem($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "delete from menu_item where codigo = " . $codigo; 
                $carga = mysqli_query ($con,$sql);
                echo "Exclusão com sucesso";
            }catch(Exception $e){
                echo "Erro de exclusão: " . $e;        
            }
        mysqli_close($con);
    }
  }
}
?>